### DISCUSSION TITLE
how to use revalidatePath properly

### DISCUSSION THREAD

**Discussion by marco-digennaro** [vercel/next.js#52488] [App Router]
hello i try to explain how im using _revalidatePath_, probably i didn't understand how to use it properly.
i simply call a server action from a client component.
the action makes just a fetch request and i call _revalidatePath(path)_ on success.
i'm using **next-intl** and i tried to pass the locale to the path but also tried to revalidate without it, but it doesn't work.

What am i doing wrong?

some pseudo code:

`
```
import { revalidatePath, useEffect } from "next/cache";
import React, { useCallback } from "react";

export default function Home() {
  const getData = useCallback(async (value?: string) => {
    const response = await fetch("endpoint");

    if (response.ok) {
      revalidatePath(`/path`);
      return response.json();
    }
  }, []);

  useEffect(() => {
    getData(value);
  }, [value]);

  return <div>page</div>;
}
```
`

Imagine that getData can be a POST where it accept an argument and the response should change the data.

**Comment by icyJoseph** [comment]
Show some code, so we can talk over it :) even pseudo-code should be fine.

**Comment by icyJoseph** [comment]
@marco-digennaro so I guess that code doesn't even work right? calling revalidate Path from the client, used to crash, and it probably still does.

This is where server actions come in handy, you'd need to define server action that calls revalidatePath, and that'll do the trick. 

For example here, https://github.com/icyJoseph/next-js-react-gbg/blob/feat/app-router-rsc/demo-app/src/app/pokemon/capture/page.tsx, I define a server action that revalidates my collection path, from the capture path. Pass that to the client, and it gets called right after I call the API that does some work for the user.

**Reply by marco-digennaro** [reply]
sorry i oversimplified my code:


`
```
"use client"

export const myFunction = () => {
 const getData = useCallback(async (value?: string) => {
    const { data } = await getMyData({ value });
  }, []);

  useEffect(() => {
    getData(value);
  }, [value]);

  return <div>page</div>;
}
```
`


`
```
"use server"

export const getMyData = async ()=>{
  const res = await fetcher(endpoint, {
    options,
  });

  if (!res) {
    throw new Error("Failed to fetch data");
  }

revalidatePath('path');

  return res;
  }
```
`

**Reply by icyJoseph** [reply]
Have you tried putting the getData on a button? and tried with wrapping with `startTransition`?
